hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -264,50 +261,71 @@ private DistributionSetAssignmentResult assignDistributionSetToTargets(final Lon
+    }
 
+    private Map<String, JpaAction> doAssignDistributionSetToTargets(
+            final Collection<TargetWithActionType> targetsWithActionType, final String actionMessage,
+            final AbstractDsAssignmentStrategy assignmentStrategy, final JpaDistributionSet distributionSetEntity,
+            final List<JpaTarget> targetEntities) {
+        final List<List<Long>> targetEntitiesIdsChunks = getTargetEntitiesAsChunks(targetEntities);
         closeOrCancelActiveActions(assignmentStrategy, targetEntitiesIdsChunks);
         // cancel all scheduled actions which are in-active, these actions were
         // not active before and the manual assignment which has been done
         // cancels them
         targetEntitiesIdsChunks.forEach(this::cancelInactiveScheduledActionsForTargets);
-
         setAssignedDistributionSetAndTargetUpdateStatus(assignmentStrategy, distributionSetEntity,
                 targetEntitiesIdsChunks);
-
-        final Map<String, JpaAction> controllerIdsToActions = createActions(targetsWithActionType, targetEntities,
+        final Map<String, JpaAction> assignedActions = createActions(targetsWithActionType, targetEntities,
                 assignmentStrategy, distributionSetEntity);
         // create initial action status when action is created so we remember
         // the initial running status because we will change the status
         // of the action itself and with this action status we have a nicer
         // action history.
-        createActionsStatus(controllerIdsToActions.values(), assignmentStrategy, actionMessage);
+        createActionsStatus(assignedActions.values(), assignmentStrategy, actionMessage);
 
         detachEntitiesAndSendTargetUpdatedEvents(distributionSetEntity, targetEntities, assignmentStrategy);
+        return assignedActions;
+    }
 
-        return new DistributionSetAssignmentResult(distributionSetEntity,
-                targetEntities.stream().map(Target::getControllerId).collect(Collectors.toList()),
-                targetEntities.size(), controllerIDs.size() - targetEntities.size(),
-                Lists.newArrayList(controllerIdsToActions.values()), targetManagement);
+    /**
+     * split tIDs length into max entries in-statement because many database
+     * have constraint of max entries in in-statements e.g. Oracle with maximum
+     * 1000 elements, so we need to split the entries here and execute multiple
+     * statements
+     */
+    private static List<List<Long>> getTargetEntitiesAsChunks(final List<JpaTarget> targetEntities) {
+        return Lists.partition(targetEntities.stream().map(Target::getId).collect(Collectors.toList()),
+                Constants.MAX_ENTRIES_IN_STATEMENT);
+    }
+
+    private static DistributionSetAssignmentResult buildAssignmentResult(final JpaDistributionSet distributionSet,
+            final Map<String, JpaAction> assignedActions, final Collection<TargetWithActionType> controllerIds) {
```

Comment:
assignedActions could be passed as JpaAction-Collection and controllerIds as int